internal/conpty/conpty.go

```diff
@@ -0,0 +1,153 @@
+package conpty
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+	"unsafe"
+
+	"github.com/Microsoft/hcsshim/internal/winapi"
+	"golang.org/x/sys/windows"
+)
+
+var (
+	errClosedConPty   = errors.New("pseudo console is closed")
+	errNotInitialized = errors.New("pseudo console hasn't been initialized")
+)
+
+// CPty is a wrapper around a Windows PseudoConsole handle. Create a new instance by calling `New()`.
+type CPty struct {
+	// handleLock guards hPc
+	handleLock sync.RWMutex
+	// hpc is the pseudo console handle
+	hpc windows.Handle
+	// inPipe and outPipe are our end of the pipes to read/write to the pseudo console.
+	inPipe  *os.File
+	outPipe *os.File
+}
+
+// New returns a new `CPty` object. This object is not ready for IO until `UpdateProcThreadAttribute` is called and a process has been started.
+func New(columns, rows int16, flags uint32) (*CPty, error) {
```

Comment:
Wonder if this would be cleaner just taking in a windows.COORD for the first two params. Same for `Resize`

Reply:
I'd have to see an example use case of the call I think

Reply:
```go
// instead of 
cpty, err := conpty.New(80, 25, 0)

// you'd have the below
coord := windows.Coord{X: 80, Y: 20}
cpty, err := conpty.New(coord, 0)
```

Reply:
yeah I guess the question is like where are these coordinates being sent from?

Reply:
In our case they'd be supplied by a client of containerd on a call to resize the task. Can see an example case in ctr https://github.com/containerd/containerd/blob/f0a32c66dad1e9de716c9960af806105d691cd78/cmd/ctr/commands/tasks/tasks_windows.go#L51

Reply:
Talked offline, just going to keep as is but rename to width/height